roles/rsyslog/defaults/main.yml

```diff
@@ -46,92 +46,29 @@ rsyslog_custom_config_files: []
 # By setting false, it'd change 2020-03-27T14:16:47.139796+00:00)
 rsyslog_basics_use_traditional_timestamp_format: true
 
-# Encrypted communication
+# communication over tls
 # ------------------------
 
-# X.509 certificates related variables
-
-# rsyslog_pki_path
-#
-# Path to the directory with PKI realms.
-rsyslog_pki_path: "/etc/pki"
-
-# rsyslog_pki_realm
-#
-# Name of the PKi realm to use with ``rsyslogd``.
-rsyslog_pki_realm: "domain"
-
-# rsyslog_pki_ca
-#
-# Name of the root CA certificate used by the ``rsyslog`` role.
-rsyslog_pki_ca: "CA.crt"
-
-# rsyslog_pki_crt
-#
-# Name of the client certificate file used by the ``rsyslog`` role.
-rsyslog_pki_crt: "default.crt"
-
-# rsyslog_pki_key
-#
-# Name of the private key file used by the ``rsyslog`` role.
-rsyslog_pki_key: "default.key"
-
-# Lookup rsyslog network stream driver
-__rsyslog_encryption_map:
-  none: ptcp
-  ptcp: ptcp
-  tls: gtls
-  gtls: gtls
-  gnutls: gtls
-  openssl: openssl
-
-# rsyslog_default_netstream_driver
-#
-# Specify the default netstream driver used by the ``imtcp`` module.
-rsyslog_default_netstream_driver: '{{ __rsyslog_encryption_map[logging_encryption] | d("ptcp") }}'
-
-# rsyslog_default_driver_authmode
-#
-# Specify the default network driver authetication mode. Actualy only
-# `x509/name` or `anon` are available:
-rsyslog_default_driver_authmode: "x509/name"
+__rsyslog_default_netstream_driver: '{{ __rsyslog_pki_map[logging_pki] | d("ptcp") }}'
 
 # rsyslog_send_over_tls
 #
 # This configuration is added to the forward options when ``tls`` capability is
 # enabled. It's used to configure TLS options.
 rsyslog_send_over_tls: |-
-  $ActionSendStreamDriver {{ rsyslog_default_netstream_driver }}
-  $ActionSendStreamDriverAuthMode {{ rsyslog_default_driver_authmode }}
-  {% if rsyslog_default_driver_authmode != "anon" %}
-  {%   if rsyslog_send_permitted_peers is string %}
-  $ActionSendStreamDriverPermittedPeer {{ rsyslog_send_permitted_peers }}
+  $ActionSendStreamDriver {{ __rsyslog_default_netstream_driver }}
+  $ActionSendStreamDriverAuthMode {{ logging_pki_authmode }}
+  {% if logging_pki_authmode != "anon" %}
```

Comment:
I think this would be better to check explicitly for auth modes that use permitted peers e.g.
```
{% if logging_pki_authmode == "x509/name" %}
```
that way we won't be surprised if we add another authmode in the future that doesn't support permitted peers.